src/faldbt/project.py

```diff
@@ -65,40 +84,234 @@ def __post_init__(self):
+
+        return res
+
+    def ref(
+        self, target_model_name: str, target_package_name: Optional[str] = None
+    ) -> pd.DataFrame:
+        """
+        Download a dbt model as a pandas.DataFrame automagically.
+        """
+
+        target_model: MaybeNonSource = self._manifest.nativeManifest.resolve_ref(
+            target_model_name, target_package_name, self.project_dir, self.project_dir
+        )
+
+        if target_model is None:
+            raise Exception(
+                f"Could not find model {target_model_name}.{target_package_name or ''}"
+            )
+
+        result = lib.fetch_target(
+            self._manifest.nativeManifest, self.project_dir, target_model
+        )
+        return pd.DataFrame.from_records(
+            result.table.rows, columns=result.table.column_names
+        )
+
+    def source(self, target_source_name: str, target_table_name: str) -> pd.DataFrame:
+        """
+        Download a dbt source as a pandas.DataFrame automagically.
+        """
+
+        target_source: MaybeNonSource = self._manifest.nativeManifest.resolve_source(
+            target_source_name, target_table_name, self.project_dir, self.project_dir
+        )
+
+        if target_source is None:
+            raise Exception(
+                f"Could not find source '{target_source_name}'.'{target_table_name}'"
+            )
+
+        result = lib.fetch_target(
+            self._manifest.nativeManifest, self.project_dir, target_source
+        )
+        return pd.DataFrame.from_records(
+            result.table.rows, columns=result.table.column_names
+        )
+
+    def write_to_source(
+        self, data: pd.DataFrame, target_source_name: str, target_table_name: str
+    ):
+        """
+        Write a pandas.DataGrame to a dbt source automagically.
+        """
+
+        target_source: MaybeParsedSource = self._manifest.nativeManifest.resolve_source(
+            target_source_name, target_table_name, self.project_dir, self.project_dir
+        )
+
+        if target_source is None:
+            raise Exception(
+                f"Could not find source '{target_source_name}'.'{target_table_name}'"
+            )
+
+        lib.write_target(
+            data,
+            self._manifest.nativeManifest,
+            self.project_dir,
+            self.profiles_dir,
+            target_source,
+        )
+
+    def write_to_firestore(self, df: pd.DataFrame, collection: str, key_column: str):
+        df_arr = df.to_dict("records")
+        for item in df_arr:
+            key = item[key_column]
+            data = _firestore_dict_to_document(data=item, key_column=key_column)
+            self.firestore_client.collection(collection).document(str(key)).set(data)
+
+    def _setup_firestore(self):
+        app_name = f"fal-{uuid.uuid4()}"
+        profile_cred = self._config.credentials
+        # Setting projectId from the profiles.yml
+        options = {"projectId": profile_cred.database}
+
+        try:
```

Comment:
Why don't we first try the profiles.yml credentials?

Reply:
I thought that it would be ideal this way. You prefer profiles first?

Reply:
It's actually debatable what makes sense to o.

Reply:
Sure. Here's an argument:

A user might have several dbt projects, each with dedicated google credentials that they specify in project-specific profiles. On top of that, they might have default credentials setup for personal use. Such user will have to unset their default credentials in order to make fal run properly, which is not a desirable workflow.

Reply:
And this is assuming a user that would have a dbt project to download the data (bigquery), would normally want to upload it to the same google project. I think that is reasonable.

Maybe a problem is how he could download data (bigquery) from one project and upload it to a different one.

This seems to be not easy to fix, but we could tackle it later.